extensions/machine-learning-services/src/packageManagement/sqlRPackageManageProvider.ts

```diff
@@ -0,0 +1,154 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+'use strict';
+
+import * as vscode from 'vscode';
+import * as azdata from 'azdata';
+import * as nbExtensionApis from '../typings/notebookServices';
+
+import { QueryRunner } from '../common/queryRunner';
+import { ApiWrapper } from '../common/apiWrapper';
+import { ProcessService } from '../common/processService';
+import { Config } from '../configurations/config';
+import { SqlPackageManageProviderBase } from './SqPackageManageProviderBase';
+
+const installMode = 'install';
+const uninstallMode = 'uninstall';
+
+/**
+ * Manage Package Provider for python packages inside SQL server databases
+ */
+export class SqlRPackageManageProvider extends SqlPackageManageProviderBase implements nbExtensionApis.IPackageManageProvider {
+
+	private _rExecutable: string;
+
+	public static ProviderId = 'sql_R';
+
+	/**
+	 * Creates new a instance
+	 */
+	constructor(
+		private _outputChannel: vscode.OutputChannel,
+		apiWrapper: ApiWrapper,
+		private _queryRunner: QueryRunner,
+		private _processService: ProcessService,
+		private _config: Config) {
+		super(apiWrapper);
+		this._rExecutable = this._config.rExecutable;
+	}
+
+	/**
+	 * Returns provider Id
+	 */
+	public get providerId(): string {
+		return SqlRPackageManageProvider.ProviderId;
+	}
+
+	/**
+	 * Returns package target
+	 */
+	public get packageTarget(): nbExtensionApis.IPackageTarget {
+		return { location: 'SQL', packageType: 'R' };
+	}
+
+	/**
+	 * Returns list of packages
+	 */
+	public async listPackages(): Promise<nbExtensionApis.IPackageDetails[]> {
+		let packages = await this._queryRunner.getRPackages(await this.getCurrentConnection());
+		if (packages) {
+			packages = packages.sort((a, b) => a.name.localeCompare(b.name));
```

Comment:
can you have multiple versions of the same package?  if so, maybe a secondary sort on version number

Reply:
yes have to do that. might do it in next iteration

---
In reply to: [366619864](https://github.com/microsoft/azuredatastudio/pull/8870#discussion_r366619864) [](ancestors = 366619864)